pydm/widgets/image.py

```diff
@@ -287,18 +287,39 @@ def image_width_changed(self, new_width):
             return
         self._image_width = int(new_width)
 
+    @staticmethod
+    def process_image(image):
+        """
+        Boilerplate method to be used by applications in order to add calculations and also modify the image before
+        it is displayed at the widget.
+
+        Parameters
+        ----------
+        image : np.ndarray
+            The Image Data as a 2D numpy array
+
+        Returns
+        -------
+        np.ndarray
+            The Image Data as a 2D numpy array after processing.
+        """
+        return image
+
     def redrawImage(self):
         """
         Set the image data into the ImageItem, if needed.
 
         If necessary, reshape the image to 2D first.
         """
+        self.redraw_timer.stop()
```

Comment:
I think it is not necessary to stop/start `redraw_timer` within `redrawImage`, because QTimer will execute `redrawImage` again only after a current execution ends, what already ensures that we process a new image only after we finished with the previous one.

Reply:
@anacso17 that is absolutely right! And this is also an issue... The QTimer is running the "processing" part in the `main thread` and this is nuking down the whole app when we add some calculation. What I am working now in a separated branch is to move the image reshape, processing and normalization to a QThread.
This will be better overall but will add a lot more of complexity to the user while writing their own code... that is why I opted to make `process_image` a static method... You can always get the ImageView widget from your code and act on the properties but you will not be allowed to touch directly QtWidgets from there...
I am updating now the `image_processing` example to see how it will come together.